src/Features/LanguageServer/Protocol/Handler/CodeLens/CodeLensResolveHandler.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeLens;
+using Newtonsoft.Json.Linq;
+using Roslyn.Utilities;
+using StreamJsonRpc;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.CodeLens;
+
+[Method(LSP.Methods.CodeLensResolveName)]
+internal sealed class CodeLensResolveHandler : ILspServiceDocumentRequestHandler<LSP.CodeLens, LSP.CodeLens>
+{
+    /// <summary>
+    /// Command name implemented by the client and invoked when the references code lens is selected.
+    /// </summary>
+    private const string ClientReferencesCommand = "roslyn.client.peekReferences";
+
+    private readonly CodeLensCache _codeLensCache;
+
+    public CodeLensResolveHandler(CodeLensCache codeLensCache)
+    {
+        _codeLensCache = codeLensCache;
+    }
+
+    public bool MutatesSolutionState => false;
+
+    public bool RequiresLSPSolution => true;
+
+    public LSP.TextDocumentIdentifier GetTextDocumentIdentifier(LSP.CodeLens request)
+        => GetCacheEntry(request).CacheEntry.TextDocumentIdentifier;
+
+    public async Task<LSP.CodeLens> HandleRequestAsync(LSP.CodeLens request, RequestContext context, CancellationToken cancellationToken)
+    {
+        var document = context.GetRequiredDocument();
+        var (cacheEntry, memberToResolve) = GetCacheEntry(request);
```

Comment:
could this fail?

Reply:
It shouldn't (but if we mess something up or the client does, it could).  It'll throw and be captured by the report NFW in the queue